### PR TITLE
Add test data for the add_keyword option

### PR DESCRIPTION
**Pull request by guy-maurel**
Ref. https://coveralls.io/builds/17772219/source?filename=src/keywords.cpp#L420

### CODE REVIEW

**Comment by mwoehlke-kitware** [comment]
Shouldn't this test actually *use* `MACRO2_BEGIN_STUFF`? Or do I miss something?

**Comment by gmaurel** [comment]
No. Have a look to` uncrustify/tests/config/custom-open.cfg`

**Comment by mwoehlke-kitware** [comment]
Hmm... I still think this should actually use the keyword. I see you are trying to exercise this code:
https://github.com/uncrustify/uncrustify/blob/25c439a13856efd39d5c08f263629da8b33df3c8/src/keywords.cpp#L421-L424

...but it still seems it would be better to follow up with a format test that tries to verify that the code actually worked as intended. In this case, I would imagine doing so by having a formatting test that works correctly if the keyword is seen as the 'updated' type, and wrongly if it is seen as the original type. (For paranoia's sake, making the original type `CT_WORD`, i.e. what uncrustify would do if there wasn't an entry at all, is probably good also.)

Actually, you could probably achieve that by sticking `word MACRO2_BEGIN_STUFF` at the beginning of `custom-open.cfg` (maybe with a comment why that's being done):

```
# Set to something differently initially, to test reassigning a custom keyword's type
word MACRO2_BEGIN_STUFF
```

**Comment by gmaurel** [comment]
I don't understand, what you wish.
The file `options.cpp` at lines 74... describes what can be done.
```.cpp
# - Custom macro-based indentation can be set up using 'macro-open',
#   'macro-else' and 'macro-close'.
#     `(macro-open | macro-else | macro-close) tokenString`
```
and at lines 2289... we can see:
```.cpp
   else if (strcasecmp(args[0], "macro-open") == 0)
   {
      add_keyword(args[1], CT_MACRO_OPEN);
```
What is `word` in your example?

**Comment by mwoehlke** [comment]
> What is `word` in your example?

Ah, I guess I'm forgetting `set`, i.e. `set WORD MACRO2_BEGIN_STUFF`. Sorry. (I thought I was remembering that we checked anything unrecognized against token types, but I must have hallucinated that.)

File: tests/config/macro-changed.cfg [hunk 1]
```diff
@@ -0,0 +1,3 @@
+indent_with_tabs = 0
+macro-open  MACRO2_BEGIN_STUFF
```
Comment (mihaipopescu) [inline tests/config/macro-changed.cfg]: have no idea what is this?

Comment (gmaurel) [reply tests/config/macro-changed.cfg]: anybody to approve?

Comment (mihaipopescu) [reply tests/config/macro-changed.cfg]: ok but you haven't answered what is `macro-open` and where is it used?

Comment (guy-maurel) [reply tests/config/macro-changed.cfg]: Sorry. macro-open and others are documentated at
option.cpp:73:
```.cpp
# - Custom macro-based indentation can be set up using 'macro-open',
#   'macro-else' and 'macro-close'.
#     `(macro-open | macro-else | macro-close) tokenString`
#
#     Example:
#       `macro-open  BEGIN_TEMPLATE_MESSAGE_MAP`
#       `macro-open  BEGIN_MESSAGE_MAP`
#       `macro-close END_MESSAGE_MAP`
```
and decoded at option.cpp:2299
```.cpp
   else if (strcasecmp(args[0], "macro-open") == 0)
   {
      add_keyword(args[1], CT_MACRO_OPEN);
   }
   else if (strcasecmp(args[0], "macro-close") == 0)
   {
      add_keyword(args[1], CT_MACRO_CLOSE);
   }
   else if (strcasecmp(args[0], "macro-else") == 0)
   {
      add_keyword(args[1], CT_MACRO_ELSE);
```

**Comment by mwoehlke-kitware** [comment]
To clarify my prior comment... what I meant was that I don't think we need an entire new test; we can achieve both goals (give coverage to replacing a custom token type, verify that the twice-set token is seen as the correct type) by leveraging one of the existing tests, as in:
```diff
diff --git a/tests/config/custom-open2.cfg b/tests/config/custom-open2.cfg
index b78ce54e..15e3710f 100644
--- a/tests/config/custom-open2.cfg
+++ b/tests/config/custom-open2.cfg
@@ -1,8 +1,10 @@
 indent_columns                  = 4
 indent_with_tabs                = 0
+# Set to something differently initially, to test reassigning a custom keyword's type
+set word                          MACRO2_BEGIN_STUFF
 macro-open                        MACRO2_BEGIN_STUFF
 macro-else                        MACRO2_ELSE_STUFF
 macro-close                       MACRO2_END_STUFF
 macro-open                        MACRO_BEGIN_STUFF
 macro-else                        MACRO_ELSE_STUFF
-macro-close                       MACRO_END_STUFF
\ No newline at end of file
+macro-close                       MACRO_END_STUFF
```

...no other changes needed.

**Pull request merged**